make/server.mk

```diff
@@ -0,0 +1,74 @@
+.PHONY: server-binaries
+server-binaries: controller acmesolver webhook cainjector
```

Comment:
On a developer laptop, we don't get much because the server binaries obviously need to be run in k8s; the idea for these targets is that they could be run in build containers and copied across in a multi-stage container build.

(put another way, this is more about defining the command required to build the container rather than it being a target we'd use for dev, although it can be useful as a command to build + check that the thing builds successfully)

Reply:
I think it is totally OK to build it outside of a container since we have no dependency on the host (no dynamically linked libraries, not even `libc.so`, and no OS-specific paths embedded thanks to `-trimpath`). The only dependency is Go itself.

At first, I thought "this is going to make the build unreproducible!!". The more I think about it, the more I think I'm OK with that since it allows for a massive speed-up while developing.

It would be nice if there was a target `binary` that would only build linux-amd64 though since that's what (probably) most people will want while developing.

Reply:
I think ideally we won't be building them locally and copying in, if only because it'll violate SLSA3's requirement for an [ephemeral environment](https://slsa.dev/requirements#ephemeral-environment) - plus there could be differences in go versions on dev laptops. It's _totally_ fine for quick development work, but the main reason this target is defined is so that it can be invoked inside a Containerfile. Any build step run anywhere should have a make target for it.

> It would be nice if there was a target binary that would only build linux-amd64 though since that's what (probably) most people will want while developing.

I think this is served by the `bin/controller-linux-amd64` target right? E.g. `make -f make/Makefile bin/controller-linux-amd64`